packages/normalizr/src/denormalize.ts

```diff
@@ -51,25 +47,33 @@ Make sure you do not have multiple versions of @rest-hooks/normalizr installed.`
     return [entity, false, false];
   }
 
+  const pk =
+    typeof entityOrId === 'string'
+      ? entityOrId
+      : schema.pk(isImmutable(entity) ? (entity as any).toJS() : entity);
+  if (pk === undefined || pk === '') {
+    return [entity, false, false];
+  }
+
   if (localCache[schema.key] === undefined) {
-    localCache[schema.key] = {};
+    localCache[schema.key] = Object.create(null);
```

Comment:
@ljharb so this should protect against prototype pollution? (remember schema.key is dev set, but pk is based on network response and we index like so `localCache[schema.key][pk]`)

Reply:
Yep! It’s safe to put any user-supplied keys into a null object.